pkg/overlay/service.go

```diff
@@ -8,40 +8,67 @@ import (
 	"flag"
 	"fmt"
 	"net"
-
+	"path/filepath"
 	"go.uber.org/zap"
 	"google.golang.org/grpc"
-	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+	"gopkg.in/spacemonkeygo/monkit.v2"
 
 	"storj.io/storj/pkg/kademlia"
 	proto "storj.io/storj/protos/overlay"
 	"storj.io/storj/storage/redis"
+	"google.golang.org/grpc/credentials"
+	"storj.io/storj/pkg/utils"
 )
 
 var (
 	redisAddress  string
 	redisPassword string
 	db            int
+	certPath      string
+	keyPath       string
 )
 
 func init() {
 	flag.StringVar(&redisAddress, "cache", "", "The <IP:PORT> string to use for connection to a redis cache")
 	flag.StringVar(&redisPassword, "password", "", "The password used for authentication to a secured redis instance")
 	flag.IntVar(&db, "db", 0, "The network cache database")
+	flag.StringVar(&certPath, "certPath", "", "TLS Certificate file")
+	flag.StringVar(&keyPath, "keyPath", "", "TLS Key file")
 }
 
 // NewServer creates a new Overlay Service Server
-func NewServer() *grpc.Server {
+func NewServer() (*grpc.Server, error) {
+	t := &utils.TlsFileOpions{
```

Comment:
would be kinda cool to do something like:
https://github.com/jetstack/kube-lego

Reply:
@coyle yea - lego is tight (https://github.com/xenolf/lego) and there is an acme server implementaiton here: https://github.com/letsencrypt/boulder; however, ACME has a tight coupling with FQDNs, which we are currently not providing so I don't think we can make an ACME based infrastructure be our primary certificate generation mechanism.

Perhaps at some point we can utilize the node lookup functionality to do name resolution to be able to sue something like ACME. I think there's greater application potential for ACME in the federation/admin-node side of things tho too rather than for every node.